L1T DQM: Modified Fat Event Filter

swang373 opened the pull request:
This PR supercedes #14463 and #14634 . This was created by checking out a CMSSW_8_0_8_patch1 release and sequentially merging in #14463 and #14634, respectively. This has been tested to compile and run in a fresh release on lxplus, and the DQM configurations were reverted to their settings for P5 as per usual.

The commit motivating the pull request is 665d73d.

The `process.source.SelectEvents` attribute is now used, being the approved method for online filtering of HLT events.

The previous HLT filter was removed, but may be manually inserted to reproduce the filtering on lxplus by following the instructions under the "HLT Fat Event Filter" subsection of DQM instructions for running on [lxplus](https://twiki.cern.ch/twiki/bin/view/Sandbox/L1TDQMModuleDev#Instructions_for_lxplus).

The rationale behind the removal is that in the offline development and testing cycle, the filter is more often than not turned off to reduce the time it takes to generate and inspect monitor elements.

A corresponding PR for 81X isn't available yet for the same rationale [explained here](https://github.com/cms-sw/cmssw/pull/14463#issuecomment-221265851).

nsmith- committed:
L1T Calo Layer1 DQM Updates

Add L1 Trigger Layer1 hw/emu validation code
(L1TdeStage2CaloLayer1)

Restructure existing DQM module and improve naming
(L1TStage2CaloLayer1)

Additional calo layer 1 dqm monitorables

swang373 committed:
Clean up and refactor merging aftermath

The previous version from March 18, 2016 was a frantic merging of the
divergent histories from @nhaubrich and @vivannguyen. Since then, the
major changes are

- The logic has been checked for conflicts between the merged features.
  No conflicts were discovered, though the logic has been considerably
refactored.

- The binning for a variety of histograms have been modified to change
  the number of bins, or the position of the bin centers.

- Switch from using EventHeader.Sector() to a shifted value of
  EventerHeader.SP_ts().

swang373 committed:
Add block to use .dat files

swang373 committed:
Added new uGMT DQM Plots

For the output:
- BX vs Muon Input Links.

For the inputs from each Track-Finder:
- Muon Multiplicity Plots for
- Processor Number vs Local Hardwire Phi

swang373 committed:
Merged swang373-dqm-CMSSW_8_0_1 from repository cms-l1t-offline

swang373 committed:
Tested CMSSW_8_0_3 merge. Fixed uGMT plot binning.

mzarucki committed:
Updated L1TStage2uGT DQM module

mzarucki committed:
Updated L1TStage2uGT DQM module: BX range

abrinke1 committed:
Removed unnecessary const in return

abrinke1 committed:
New EMTF DAQ output as of April 1

swang373 committed:
Merge pull request #225 from mzarucki/uGT_DQM_v3_803

Updated L1TStage2uGT DQM module

EsmaeelEskandari committed:
BMTF is revised and OMTF is added

EsmaeelEskandari committed:
some bugs was fixed

swang373 committed:
Updates to EMTF DataFormats

The EMTF DataFormats were updated:
- Added methods which return pointers to the collections in EMTFOutput.
- A discussion with @davidlange6 and @abrinke1 has clarified the usage
  of const. Only the methods which return references or pointers will be
returned as const (because they must).

vivannguyen committed:
DQM wire/strip vs chamber, mode, and quality plots

Addition of Wire/Strip vs Chamber plots for all Stations/Rings, including Station 1. Also, addition of Mode, Quality, and Mode vs Quality plots.

abrinke1 committed:
Merged l1t-dqm-CMSSW_8_0_3 from repository cms-l1t-offline

abrinke1 committed:
EMTF unpacker changes for new firmware, April 1 and April 8

swang373 committed:
Added paths to more recent runs for test

abrinke1 committed:
Fixes to buggy unpacker update

swang373 committed:
Merge Andrew's fixes to EMTF Unpacker Bugs

swang373 committed:
Updates to EMTF DQM Module

- Reworked the logic for declaring and filling the strip and wire plots
  to take advantage of new convenience variables from the unpacker.
- Added Monitor Elements for the output RegionalMuonCandBxCollection.
- Added plots for number of tracks per event and number of tracks per
  sector processor, with an explicit overflow bin.

swang373 committed:
Merge pull request #229 from EsmaeelEskandari/l1t-dqm-CMSSW_8_0_3

L1t dqm cmssw 8 0 3

swang373 committed:
Update Phi Binning for BMTF DQM Module

nhaubrich committed:
Fixed Y-axis binning on emtfLCTBX, added HQ Eta histogram to EMTF, EMTF now gets sector from ME and SP collections instead of EventHeader

nsmith- committed:
Stage 2 Trigger DQM Quality Test framework

Still needs to be filled in by L1 subsystem experts
Also need to update L1 object definitions

nsmith- committed:
L1TCaloLayer1 DQM

Small TCC unpacker handling update + de update

nhaubrich committed:
Removed SP_ts() from unpacker and moved the phi_GMT_global to radian
conversion to the unpacker as phi_GMT_global_rad()

swang373 committed:
Move Stage2 unpacking sequence to L1TStage2_cff.py

The Stage2 unpacking sequence has been moved into L1TStage2_cff for
clarity, as that is where they are initially imported for use. There
were minor edits to variable names in the EMTF DQM module.

nsmith- committed:
DQM L1 Calo Layer 1 Quality Tests

Thresholds may need alteration

swang373 committed:
First pass at data-emulator comparison in DQM

A first pass at a Python configuration file for the data to emulator
comparison for online DQM has been made, with an emulator DQM module for
EMTF as a test case.

A few files have had executable permissions removed.

nsmith- committed:
L1 Calo Layer 1 Unpacker update

To coincide with firmware change that increases link error information
granularity in readout.

swang373 committed:
Additional DQM Plots for uGMT

Output muon phi plots with respect to each subsystem.

nsmith- committed:
Add L1 Emulator DQM sequence

Only CaloLayer1 at the moment

nsmith- committed:
Merge pull request #258 from nhaubrich/nickemtfdev803pr

Minor Changes to EMTF Module and Unpacker

nsmith- committed:
L1 DQM Guard against no HLT Fat Event filter

Will produce empty plots though

nsmith- committed:
Merged refs/pull/14141/head from repository cms-sw

nsmith- committed:
L1 CaloLayer1 data/emu DQM patch

Temporary patch for issue with Layer 2 unpacker

nsmith- committed:
Merge remote-tracking branch 'cms-l1t-offline/l1t-dqm-CMSSW_8_0_3' into l1t-dqm-CMSSW_8_0_5

Resolved Conflicts:
	DQM/Integration/python/clients/l1tstage2_dqm_sourceclient-live_cfg.py
	DQM/Integration/python/clients/l1tstage2emulator_dqm_sourceclient-live_cfg.py
	DQM/L1TMonitor/python/L1TStage2Emulator_cff.py
	DQM/L1TMonitor/python/L1TStage2_cff.py

nsmith- committed:
L1 DQM Fix loose ends from merge

Correct plot naming for EMTF emulator
Move hardware validation sequence to L1TStage2Emulator_cff for now

nsmith- committed:
L1 CaloLayer2 emu DQM

Jingyu already setup the configurations, was just a matter of plugging them together.
Also had to make valCaloStage2Layer2Digis, implementation seemed clear.

nsmith- committed:
Add back uHTR location mismatch plots to L1 CaloLayer1 DQM

nsmith- committed:
L1 DQM Undo change to es_prefer

Should not have changed this file, sorry.
Also moved calostage2 plugin back to data client for time being.
Emulator client will need more investigation related to
CaloParams/GlobalTag conflict.

nsmith- committed:
Merged l1t-dqm-CMSSW_8_0_5 from repository cms-l1t-offline

Into CMSSW_8_0_7
No conflicts

EsmaeelEskandari committed:
CaloLayer2 and BMTF are updated

EsmaeelEskandari committed:
CaloLayer2 is folder-ized and a new BMTF DQM plot is added

nhaubrich committed:
Merged refs/pull/14373/head from repository cms-sw

nhaubrich committed:
Fixed csc_id numbering in DataFormats, EMTF once again gets sector from EventHeader (instead of ME or SP), and chamber numbering errors were corrected

swang373 committed:
EMTF DQM CSC Chamber Numbering Bugfix

The chamber numbering scheme range has been fixed such that it falls
within the range 1 to 36 (1 to 18 in ring 1 for stations 2 and above),
rather than 0 to 35 (0 to 17, in ring 1 for stations 2 and above).

swang373 committed:
Merge branch 'l1t-dqm-CMSSW_8_0_7' into EMTF_Bugfix-l1t-dqm-CMSSW_8_0_7

Merge the EMTF bugfixes made to a branch based off of cms-sw/cmssw#14373 into a branch based off of cms-sw/cmssw#14376.

mzarucki committed:
uGT DQM: Algo bit vs global BX fix

swang373 committed:
Merged refs/pull/14463/head from repository cms-sw

swang373 committed:
Merged refs/pull/14634/head from repository cms-sw

cmsbuild commented:
A new Pull Request was created by  @swang373 (Sean-Jiun Wang) for CMSSW_8_0_X.

It involves the following packages:

DQM/Integration
DQM/L1TMonitor
DQM/L1TMonitorClient
DataFormats/L1TMuon
EventFilter/L1TRawToDigi
EventFilter/L1TXRawToDigi


@dmitrijus, @cmsbuild, @rekovic, @deguio, @vanbesien, @mulhearn, @davidlange6 can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @threus, @batinkov this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

swang373 commented on DQM/Integration/python/clients/l1tstage2_dqm_sourceclient-live_cfg.py:
```diff
@@ -8,6 +8,8 @@
 # Live Online DQM in P5
 process.load("DQM.Integration.config.inputsource_cfi")
 process.load("DQM.Integration.config.FrontierCondition_GT_cfi")
+process.source.SelectEvents = cms.untracked.vstring('HLT_L1FatEvents_v*')
```
This is the line to comment in order to turn off the HLT fat event filtering for L1T DQM!

swang373 committed:
L1T DQM: Toggle OFF HLT Fat Event Filter

The manually created HLT filter for fat events has been removed.
Instead, the SelectEvents member of the DQMStreamerReader object
will be used for online DQM at P5.

swang373 commented on DQM/Integration/python/clients/l1tstage2_dqm_sourceclient-live_cfg.py:
```diff
@@ -8,6 +8,8 @@
 # Live Online DQM in P5
 process.load("DQM.Integration.config.inputsource_cfi")
 process.load("DQM.Integration.config.FrontierCondition_GT_cfi")
+#process.source.SelectEvents = cms.untracked.vstring('HLT_L1FatEvents_v*')
```
This is the line to uncomment to turn ON the HLT fat event filter!

vanbesien commented:
Please close this PR.
